src/lib/mongoBackend/safeMongo.cpp

```diff
@@ -288,15 +327,52 @@ bool getBoolField(const BSONObj& b, const std::string& field, const std::string&
   }
   else
   {
-    LM_E(("Runtime Error (field '%s' was supposed to be a bool but type=%d in BSONObj <%s> from caller %s:%d)",
-          field.c_str(), b.getField(field).type(), b.toString().c_str()));
+    LM_E(("Runtime Error (field '%s' was supposed to be a bool but the type is '%s' in BSONObj <%s> from caller %s:%d)",
+          field.c_str(), mongoTypeName(b.getField(field).type()), b.toString().c_str()));
   }
 
   return false;
 }
 
 
 
+/* ****************************************************************************
+*
+* getNumberFieldAsDouble -
+*/
+double getNumberFieldAsDouble(const BSONObj& b, const std::string& field, const std::string& caller, int line)
```

Comment:
What the different between this new function and existing `double getNumberField()` ?

Reply:
This new function accepts int32, int64 or double in mongo and returns a double.
Does getNumberField do that?
If it does, then no difference whatsoever.
I will check - didn't imagine a function like that would exist already :)

Reply:
So, looked it up.
getNumberField only accepts double. Any other type in mongo will result in an error.
I'd suggest a change in the name of the function to "getDoubleField".
Not really for this PR though

NTC

Reply:
I'll have a look to both functions once this PR gets merged. I have the "feeling" that they are so close they could be unified, but not sure of it.

As you said, not in this PR.